src/db/entity/achievementProgress.ts

```diff
@@ -1,176 +1,175 @@
-import { Entity, ManyToOne, PrimaryColumn, Column } from "typeorm";
-import { User } from "./";
-import { Achievement } from "../../util/achievements";
+import { Entity, ManyToOne, PrimaryColumn, Column } from 'typeorm';
+import { User } from './';
+import { Achievement } from '../../util/achievements';
 
 /**
  * A class to store a user's progress on a specific achievement
  */
 @Entity()
 export class AchievementProgress {
-  /**
+	/**
    * The id of the achievement
    */
-  @PrimaryColumn()
-  achievementId: number;
+	@PrimaryColumn()
+	public achievementId!: number;
 
-  /**
+	/**
    * The user
    */
-  @ManyToOne(type => User, user => user.achievementsProgress, {
-    eager: true,
-    primary: true
-  })
-  user: User;
+	@ManyToOne(() => User, user => user.achievementsProgress, {
+		eager: true,
+		primary: true
+	})
+	public user!: User;
 
-  /**
+	/**
    * The progress that has been made
    */
-  @Column({
-    nullable: false
-  })
-  progress: number;
+	@Column({
+		nullable: false
+	})
+	public progress!: number;
 
-  /**
+	/**
    * All completed steps stored in a JSON array as a string
    */
-  @Column({
-    nullable: false,
-    type: "simple-json"
-  })
-  completedSteps: number[];
+	@Column({
+		nullable: false,
+		type: 'simple-json'
+	})
+	public completedSteps!: number[];
 
-  /**
+	/**
    * Wether or not the user has claimed their prize for the achievement
    */
-  @Column()
-  prizeClaimed: boolean;
+	@Column()
+	public prizeClaimed!: boolean;
 
-  /**
+	/**
    * The achievement (not stored on the database)
    */
-  achievement: Achievement;
+	public achievement!: Achievement;
 
-  /**
+	/**
    *
    * @param achievement The achievement
    * @param user The user
    * @param progress (optional) The progress the user has made on this achievement
    * @param stepsCompleted (optional) The steps the user has completed
    * @param prizeClaimed (optional) Wether or not the user has claimed their prize for the achievement
    */
-  constructor(achievement: Achievement, user: User, progress?: number, stepsCompleted?: number[], prizeClaimed?: boolean) {
-    this.achievementId = achievement ? achievement.getId() : undefined;
-    this.user = user;
-    this.progress = progress || 0;
-    this.completedSteps = stepsCompleted || [];
-    this.prizeClaimed = prizeClaimed || false;
-    this.achievement = achievement;
-  }
+	public constructor(achievement: Achievement, user: User, progress?: number, stepsCompleted?: number[], prizeClaimed?: boolean) {
+		this.achievementId = achievement.getId();
```

Comment:
For some reason, this is causing problems on startup, apparently `achievement` can be undefined? I didn't look into it too much but the suggestion below fixes the problem.
```suggestion
		this.achievementId = achievement?.getId();
```